src/dmd/statementsem.d

```diff
@@ -250,6 +250,12 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
                 (*cs.statements)[i] = s;
                 if (s)
                 {
+                    if (s.isErrorStatement())
+                    {
+                        result = s;     // propagate error up the AST
+                        ++i;
```

Comment:
this looks very suspicious to me.

Reply:
it's correct. I don't know what else to say. See how the loop is implemented.

Reply:
so we only get the last error?

Reply:
couldn't you break?